neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -128,32 +131,59 @@ class NeuralynxRawIO(BaseRawIO):
         ("channel_id", "uint32"),
         ("sample_rate", "uint32"),
         ("nb_valid", "uint32"),
-        ("samples", "int16", (NcsSection._RECORD_SIZE)),
+        ("samples", "int16", NcsSection._RECORD_SIZE),
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
+            include_filenames=None,
+            exclude_filenames=None,
+            keep_original_times=False,
+            strict_gap_mode=True,
+            filename=None,
+            exclude_filename=None,
+            **kargs
     ):
 
-        if dirname != "":
-            self.dirname = dirname
-            self.rawmode = "one-dir"
-        elif filename != "":
-            self.filename = filename
-            self.rawmode = "one-file"
+        if not dirname:
+            raise ValueError("`dirname` cannot be empty.")
+
+        if filename is not None:
+            include_filenames = [filename]
+            warnings.warn("`filename` is deprecated and will be removed. Please use `include_filenames` instead")
+
+        if exclude_filename is not None:
+            if isinstance(exclude_filename, str):
+                exclude_filenames = [exclude_filename]
+            else:
+                exclude_filenames = exclude_filename
+            warnings.warn("`exclude_filename` is deprecated and will be removed. Please use `exclude_filenames` instead")
+
+        if include_filenames is None:
+            include_filenames = []
+        elif isinstance(include_filenames, str):
+            include_filenames = [include_filenames]
```

Comment:
This is the one I'm still a little torn on. I would prevent to force include_filenames to be a list, but I think it is okay if we accept a string in the case of only wanting one file.

Reply:
Under the hood, `include_filenames` has to be a list, or the iteration at line 220 would loop over each string character.

Reply:
Sorry I wasn't clear. I was saying why should we accept a string at all (so delete line 165 and 166). Since our new argument is filename`s` shouldn't we just always make the users give a list (or any iterable)? Why should we accept a single file rather than a list of a single file?

Reply:
I asked on Chatgpt and it seems quite common to allow for flexible input types in this case. I am open to both ways though.

Here is the answer from chatgpt:

In Python, when designing an I/O interface to read a list of files, it's generally user-friendly to allow for flexible input types. Here are the considerations for both approaches you mentioned:

Allowing a string or list as input:

Pros: More convenient for users who often deal with a single file. It reduces the overhead of wrapping a filename in a list, making the function easier to use in scripts or interactive sessions where only one file is processed.
Cons: The function implementation needs to handle different types of input, checking whether the input is a string or a list. This can make the function slightly more complex.
Strictly requiring a list of strings:

Pros: Consistency in input type can simplify the function's logic, as it always expects a list. This can make the code simpler and potentially less error-prone when dealing with multiple files.
Cons: Less convenient for cases where only a single file is involved, as the user must wrap the filename in a list even if it's just one.
Given these considerations, a common practice is to design the interface to accept both strings and lists. This approach enhances usability without significantly complicating the implementation.

Reply:
@apdavison did you have opinions on this? I prefer to be more restrictive, but I know other parts of Neo's API is relatively permissive. I think you prefer more flexibility too @samuelgarcia, so I'm fine with being outvoted here.

Reply:
I think it's a tough call, but I can't see that accepting a string will cause major problems, so I'm happy to go with the current interface.